File: src/utils/functions/power.js
// import {abs} from './abs.js'
// import {ePower} from './ePower.js'

// export function power (num, exponent) {
//   let result = 1
//   let absExponent = abs(exponent)
//   if (absExponent % 1 !== 0) {
//     let decimalExponent = absExponent % 1
//     let wholeExponent = absExponent - decimalExponent
//     result = power(num, wholeExponent) * ePower(decimalExponent * myln(num))
//   } else {
//     for (let i = 1; i <= absExponent; ++i) {
//       result = result * num
//     }
//   }

//   if (exponent < 0) {
//     return 1 / result
//   } else {
//     return result
//   }
// }

// export function myln (num) {
//   let rounds = 300
//   let temp = ((num - 1) / (num + 1))
//   let result = 0
//   for (let i = 1; i < rounds * 2; i += 2) {
//     let j = 1 / i
//     result = result + (j * power(temp, i))
//   }
//   return (2.0 * result)
// }

File: app/functions/power.py
from .abs import abs
from .ePower import ePower


def power(num, exponent):
  result = 1
  abs_exponent = abs(exponent)
  if abs_exponent % 1 != 0:
    decimal_exponent = abs_exponent % 1
    whole_exponent = int(abs_exponent)
    result = power(num, whole_exponent) * ePower(decimal_exponent * my_ln(num))
  else:
    abs_exponent = int(abs_exponent)
    for i in range(1, abs_exponent + 1):
      result = result * num

  if exponent < 0:
    return 1 / result
  else:
    return result


def my_ln(num):
  rounds = 300
  temp = ((num - 1) / (num + 1))
  result = 0
  for i in range(1, rounds * 2, 2):
    j = 1 / i
    result = result + (j * power(temp, i))
  return (2.0 * result)
